Source/VideoProgressView.swift

```diff
@@ -0,0 +1,119 @@
+        maskView.clipsToBounds = true
+        maskView.layer.masksToBounds = true;
+        return maskView
+    }()
+
+
+    lazy var backgroundBar: UIView = {
+        let backgroundBar = UIView()
+        backgroundBar.backgroundColor = UIColor.whiteColor()
+        backgroundBar.alpha = 0.2
+        return backgroundBar
+    }()
+
+    lazy var progressBar: UIView = {
+        let progressBar = UIView()
+        progressBar.backgroundColor = UIColor.whiteColor()
+        return progressBar
+    }()
+
+    lazy var currentTimeLabel: UILabel = {
+        let currentTimeLabel = UILabel()
+        currentTimeLabel.font = UIFont(name: "DINNextLTPro-Regular", size: 14)
+        currentTimeLabel.textColor = UIColor.whiteColor()
+        return currentTimeLabel
+    }()
+
+    lazy var durationTimeLabel : UILabel = {
+        let durationTimeLabel = UILabel()
+        durationTimeLabel.font = UIFont(name: "DINNextLTPro-Regular", size: 14)
+        durationTimeLabel.textColor = UIColor.whiteColor()
+        return durationTimeLabel
+    }()
+
+    override init(frame: CGRect) {
+        
+        super.init(frame: frame)
+
+        self.addSubview(self.maskBarForRoundedCorners)
+        self.maskBarForRoundedCorners.addSubview(self.backgroundBar)
+        self.maskBarForRoundedCorners.addSubview(self.progressBar)
+
+        self.addSubview(self.currentTimeLabel)
+        self.addSubview(self.durationTimeLabel)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func layoutSubviews(){
+        super.layoutSubviews()
+
+        self.maskBarForRoundedCorners.frame = CGRect(x: VideoProgressView.ProgressBarXMargin, y: VideoProgressView.ProgressBarYMargin, width: self.widthForBar, height: VideoProgressView.ProgressBarHeight)
+        self.backgroundBar.frame = CGRect(x: 0, y: 0, width: self.widthForBar, height: VideoProgressView.ProgressBarHeight)
+        self.setFrameForProgressBar()
+
+        let xPosForCurrentTimeLabel = (VideoProgressView.ProgressBarXMargin - VideoProgressView.TextLabelWidth)/2
+        self.currentTimeLabel.frame = CGRect(x: xPosForCurrentTimeLabel, y: VideoProgressView.TextLabelMargin, width: VideoProgressView.TextLabelWidth, height: VideoProgressView.TextLabelHeight)
+        let xPosForDurationTimeLabel = self.bounds.width - VideoProgressView.TextLabelWidth - (VideoProgressView.ProgressBarXMargin - VideoProgressView.TextLabelWidth)/2
+        self.durationTimeLabel.frame = CGRect(x: xPosForDurationTimeLabel, y: VideoProgressView.TextLabelMargin, width: VideoProgressView.TextLabelWidth, height: VideoProgressView.TextLabelHeight)
+    }
+
+    func setFrameForProgressBar(){
+        self.progressBar.frame = CGRect(x: 0, y: 0, width: self.widthForProgressBar, height: VideoProgressView.ProgressBarHeight)
+    }
+
+    func timeStringForSeconds(secondValue : Double) -> String{
+       let minutes : Int = Int((secondValue % 3600) / 60)
```

Comment:
The type declaration is not needed here since it gets inferred on declaration.

This should be enough:

```swift
let minutes = Int((secondValue % 3600) / 60)
```